### PR TITLE
CUDA and requirements

### PR DESCRIPTION
**Pull request by ashutosh1919**
Since we know that installing CUDA on newly created instance is a complex tasks because there will be many errors related to version conflicts. Especially while installing NVIDIA drivers and cuDNN. Also, it is a big headache downloading files in cloud instance. I have added **CUDA_INSTALLATION_INSTRUCTION.md** file which helps users in installing CUDA and setting up environment to run stylegan project. I have tested this on **Azure**, **AWS** and **GCP**. Works fine. Below md file explains everything about the installation. 
Also added requirements.txt file since it's useful while creating virtualenv on cloud VM instance.
@tkarras, Please review this PR. And reply if you find it useful.

### CODE REVIEW

File: CUDA_INSTALLATION_INSTRUCTIONS.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+Connect the VM instance through terminal. (If the cloud you are using don't provide terminal then use [puTTY](https://www.ssh.com/ssh/putty)).
+
+## Installing NVIDIA Graphics Drivers
+Note here that CUDA-9.0 supports NVIDIA Drivers with version 384 and above.
+```sh
+$ sudo apt-get update
+$ sudo apt-get install nvidia-384 nvidia-modprobe
+$ sudo apt-get update
+```
+After executing above commands, you can check weather the installation is complete or not with the command `nvidia-smi`. If the installation is complete then it will show the table corresponding to particular NVIDIA Driver.
```
Comment (yayajacky) [inline CUDA_INSTALLATION_INSTRUCTIONS.md]: weather -> whether

Comment (ashutosh1919) [reply CUDA_INSTALLATION_INSTRUCTIONS.md]: done

File: CUDA_INSTALLATION_INSTRUCTIONS.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+Downloading the cuDNN file from [NVIDIA cuDNN website](https://developer.nvidia.com/rdp/cudnn-download) can be very frustrating if you are using cloud VM through terminal because you can only download the file to your computer. You can't download it in VM by `wget` because it changes the configuration of `.deb` or `.tgz` file. 
+But the solution here is we can download raw cuDNN `.deb` files from [NVIDIA Repo](http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1604/x86_64/) like we download raw `.whl` files for pip installing package.
+You can choose the link corresponding to the version of your preference.
+  - Installing runtime library first
+    ```sh
+    $ wget http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1604/x86_64/libcudnn7_7.1.3.16-1+cuda9.0_amd64.deb
+    $ chmod +x libcudnn7_7.1.3.16-1+cuda9.0_amd64.deb
+    $ sudo -dpkg -i libcudnn7_7.1.3.16-1+cuda9.0_amd64.deb
```
Comment (sqiprasanna) [inline CUDA_INSTALLATION_INSTRUCTIONS.md:87]: It's sudo dpkg -i libcudnn7_7.1.3.16-1+cuda9.0_amd64.deb (remove - from -dpkg)

**Comment by Raghvender1205** [comment]
I used <code>tf.disable_v2_behaviour()</code> function in TF = 2.3.1 to use this repo as it was giving all errors on <code>tf.Session</code> <code>tf.variable_scope.</code> After that now it is giving This Error

I ran this python command 
```
python pretrained_example.py
```

I got this Error
```
WARNING:tensorflow:From D:\ML\NVIDIA_Research\stylegan\dnnlib\tflib\tfutil.py:12: The name tf.disable_v2_behavior is deprecated. Please use tf.compat.v1.disable_v2_behavior instead.

WARNING:tensorflow:From C:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\compat\v2_compat.py:61: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
Instructions for updating:
non-resource variables are not supported in the long term
WARNING:tensorflow:From D:\ML\NVIDIA_Research\stylegan\dnnlib\tflib\tfutil.py:96: The name tf.get_default_session is deprecated. Please use tf.compat.v1.get_default_session instead.

WARNING:tensorflow:From D:\ML\NVIDIA_Research\stylegan\dnnlib\tflib\tfutil.py:108: The name tf.set_random_seed is deprecated. Please use tf.compat.v1.set_random_seed instead.

WARNING:tensorflow:From D:\ML\NVIDIA_Research\stylegan\dnnlib\tflib\tfutil.py:131: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.

WARNING:tensorflow:From D:\ML\NVIDIA_Research\stylegan\dnnlib\tflib\tfutil.py:141: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.

Downloading https://drive.google.com/uc?id=1MEGjdvVpUsu1jB4zrXZN7Y4kBBOzizDQ .... done
WARNING:tensorflow:From <string>:364: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where

Gs                              Params    OutputShape          WeightShape
---                             ---       ---                  ---
latents_in                      -         (?, 512)             -
labels_in                       -         (?, 0)               -
lod                             -         ()                   -
dlatent_avg                     -         (512,)               -
G_mapping/latents_in            -         (?, 512)             -
G_mapping/labels_in             -         (?, 0)               -
G_mapping/PixelNorm             -         (?, 512)             -
G_mapping/Dense0                262656    (?, 512)             (512, 512)
G_mapping/Dense1                262656    (?, 512)             (512, 512)
G_mapping/Dense2                262656    (?, 512)             (512, 512)
G_mapping/Dense3                262656    (?, 512)             (512, 512)
G_mapping/Dense4                262656    (?, 512)             (512, 512)
G_mapping/Dense5                262656    (?, 512)             (512, 512)
G_mapping/Dense6                262656    (?, 512)             (512, 512)
G_mapping/Dense7                262656    (?, 512)             (512, 512)
G_mapping/Broadcast             -         (?, 18, 512)         -
G_mapping/dlatents_out          -         (?, 18, 512)         -
Truncation                      -         (?, 18, 512)         -
G_synthesis/dlatents_in         -         (?, 18, 512)         -
G_synthesis/4x4/Const           534528    (?, 512, 4, 4)       (512,)
G_synthesis/4x4/Conv            2885632   (?, 512, 4, 4)       (3, 3, 512, 512)
G_synthesis/ToRGB_lod8          1539      (?, 3, 4, 4)         (1, 1, 512, 3)
G_synthesis/8x8/Conv0_up        2885632   (?, 512, 8, 8)       (3, 3, 512, 512)
G_synthesis/8x8/Conv1           2885632   (?, 512, 8, 8)       (3, 3, 512, 512)
G_synthesis/ToRGB_lod7          1539      (?, 3, 8, 8)         (1, 1, 512, 3)
G_synthesis/Upscale2D           -         (?, 3, 8, 8)         -
G_synthesis/Grow_lod7           -         (?, 3, 8, 8)         -
G_synthesis/16x16/Conv0_up      2885632   (?, 512, 16, 16)     (3, 3, 512, 512)
G_synthesis/16x16/Conv1         2885632   (?, 512, 16, 16)     (3, 3, 512, 512)
G_synthesis/ToRGB_lod6          1539      (?, 3, 16, 16)       (1, 1, 512, 3)
G_synthesis/Upscale2D_1         -         (?, 3, 16, 16)       -
G_synthesis/Grow_lod6           -         (?, 3, 16, 16)       -
G_synthesis/32x32/Conv0_up      2885632   (?, 512, 32, 32)     (3, 3, 512, 512)
G_synthesis/32x32/Conv1         2885632   (?, 512, 32, 32)     (3, 3, 512, 512)
G_synthesis/ToRGB_lod5          1539      (?, 3, 32, 32)       (1, 1, 512, 3)
G_synthesis/Upscale2D_2         -         (?, 3, 32, 32)       -
G_synthesis/Grow_lod5           -         (?, 3, 32, 32)       -
G_synthesis/64x64/Conv0_up      1442816   (?, 256, 64, 64)     (3, 3, 512, 256)
G_synthesis/64x64/Conv1         852992    (?, 256, 64, 64)     (3, 3, 256, 256)
G_synthesis/ToRGB_lod4          771       (?, 3, 64, 64)       (1, 1, 256, 3)
G_synthesis/Upscale2D_3         -         (?, 3, 64, 64)       -
G_synthesis/Grow_lod4           -         (?, 3, 64, 64)       -
G_synthesis/128x128/Conv0_up    426496    (?, 128, 128, 128)   (3, 3, 256, 128)
G_synthesis/128x128/Conv1       279040    (?, 128, 128, 128)   (3, 3, 128, 128)
G_synthesis/ToRGB_lod3          387       (?, 3, 128, 128)     (1, 1, 128, 3)
G_synthesis/Upscale2D_4         -         (?, 3, 128, 128)     -
G_synthesis/Grow_lod3           -         (?, 3, 128, 128)     -
G_synthesis/256x256/Conv0_up    139520    (?, 64, 256, 256)    (3, 3, 128, 64)
G_synthesis/256x256/Conv1       102656    (?, 64, 256, 256)    (3, 3, 64, 64)
G_synthesis/ToRGB_lod2          195       (?, 3, 256, 256)     (1, 1, 64, 3)
G_synthesis/Upscale2D_5         -         (?, 3, 256, 256)     -
G_synthesis/Grow_lod2           -         (?, 3, 256, 256)     -
G_synthesis/512x512/Conv0_up    51328     (?, 32, 512, 512)    (3, 3, 64, 32)
G_synthesis/512x512/Conv1       42112     (?, 32, 512, 512)    (3, 3, 32, 32)
G_synthesis/ToRGB_lod1          99        (?, 3, 512, 512)     (1, 1, 32, 3)
G_synthesis/Upscale2D_6         -         (?, 3, 512, 512)     -
G_synthesis/Grow_lod1           -         (?, 3, 512, 512)     -
G_synthesis/1024x1024/Conv0_up  21056     (?, 16, 1024, 1024)  (3, 3, 32, 16)
G_synthesis/1024x1024/Conv1     18752     (?, 16, 1024, 1024)  (3, 3, 16, 16)
G_synthesis/ToRGB_lod0          51        (?, 3, 1024, 1024)   (1, 1, 16, 3)
G_synthesis/Upscale2D_7         -         (?, 3, 1024, 1024)   -
G_synthesis/Grow_lod0           -         (?, 3, 1024, 1024)   -
G_synthesis/images_out          -         (?, 3, 1024, 1024)   -
G_synthesis/lod                 -         ()                   -
G_synthesis/noise0              -         (1, 1, 4, 4)         -
G_synthesis/noise1              -         (1, 1, 4, 4)         -
G_synthesis/noise2              -         (1, 1, 8, 8)         -
G_synthesis/noise3              -         (1, 1, 8, 8)         -
G_synthesis/noise4              -         (1, 1, 16, 16)       -
G_synthesis/noise5              -         (1, 1, 16, 16)       -
G_synthesis/noise6              -         (1, 1, 32, 32)       -
G_synthesis/noise7              -         (1, 1, 32, 32)       -
G_synthesis/noise8              -         (1, 1, 64, 64)       -
G_synthesis/noise9              -         (1, 1, 64, 64)       -
G_synthesis/noise10             -         (1, 1, 128, 128)     -
G_synthesis/noise11             -         (1, 1, 128, 128)     -
G_synthesis/noise12             -         (1, 1, 256, 256)     -
G_synthesis/noise13             -         (1, 1, 256, 256)     -
G_synthesis/noise14             -         (1, 1, 512, 512)     -
G_synthesis/noise15             -         (1, 1, 512, 512)     -
G_synthesis/noise16             -         (1, 1, 1024, 1024)   -
G_synthesis/noise17             -         (1, 1, 1024, 1024)   -
images_out                      -         (?, 3, 1024, 1024)   -
---                             ---       ---                  ---
Total                           26219627

Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\client\session.py", line 1356, in _do_call
    return fn(*args)
  File "C:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\client\session.py", line 1341, in _run_fn
    options, feed_dict, fetch_list, target_list, run_metadata)
  File "C:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\client\session.py", line 1429, in _call_tf_sessionrun
    run_metadata)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Cannot assign a device for operation Gs/_Run/Gs/latents_in: {{node Gs/_Run/Gs/latents_in}}was explicitly assigned to /device:GPU:0 but available devices are [ /job:localhost/replica:0/task:0/device:CPU:0 ]. Make sure the device specification refers to a valid device. The requested device appears to be a GPU, but CUDA is not enabled.
         [[Gs/_Run/Gs/latents_in]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ".\pretrained_example.py", line 47, in <module>
    main()
  File ".\pretrained_example.py", line 39, in main
    images = Gs.run(latents, None, truncation_psi=0.7, randomize_noise=True, output_transform=fmt)
  File "D:\ML\NVIDIA_Research\stylegan\dnnlib\tflib\network.py", line 443, in run
    mb_out = tf.get_default_session().run(out_expr, dict(zip(in_expr, mb_in)))
  File "C:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\client\session.py", line 950, in run
    run_metadata_ptr)
  File "C:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\client\session.py", line 1173, in _run
    feed_dict_tensor, options, run_metadata)
  File "C:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\client\session.py", line 1350, in _do_run
    run_metadata)
  File "C:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\client\session.py", line 1370, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Cannot assign a device for operation Gs/_Run/Gs/latents_in: node Gs/_Run/Gs/latents_in (defined at D:\ML\NVIDIA_Research\stylegan\dnnlib\tflib\network.py:218) was explicitly assigned to /device:GPU:0 but available devices are [ /job:localhost/replica:0/task:0/device:CPU:0 ]. Make sure the device specification refers to a valid device. The requested device appears to be a GPU, but CUDA is not enabled.
         [[Gs/_Run/Gs/latents_in]]

Errors may have originated from an input operation.
Input Source operations connected to node Gs/_Run/Gs/latents_in:
 Gs/_Run/split (defined at D:\ML\NVIDIA_Research\stylegan\dnnlib\tflib\network.py:404)
```

**Pull request closed without merging**